csharp/src/Microsoft.ML.OnnxRuntime/DisposableNamedOnnxValue.cs

```diff
@@ -125,15 +125,108 @@ internal static DisposableNamedOnnxValue CreateFromOnnxValue(string name, IntPtr
+                        NativeApiStatus.VerifySuccess(NativeMethods.OrtGetValue(nativeOnnxValue, 0, allocator, out nativeOnnxValueSeq));
+                        sequence.Add(CreateFromOnnxValue(string.Empty, nativeOnnxValueSeq, allocator));
+                        NativeMethods.OrtReleaseValue(nativeOnnxValueSeq);
+                    }
+                    return new DisposableNamedOnnxValue(name, sequence, null);
+
+                case OnnxType.ONNX_TYPE_MAP:
+                    IntPtr typeAndShape = IntPtr.Zero;
+                    IntPtr nativeOnnxValueMapKeys = IntPtr.Zero;
+                    IntPtr nativeOnnxValueMapValues = IntPtr.Zero;
+                    TensorElementType elemType = TensorElementType.DataTypeMax;
+                    NativeApiStatus.VerifySuccess(NativeMethods.OrtGetValue(nativeOnnxValue, 0, allocator, out nativeOnnxValueMapKeys));
+                    NativeApiStatus.VerifySuccess(NativeMethods.OrtGetValue(nativeOnnxValue, 1, allocator, out nativeOnnxValueMapValues));
+                    NativeApiStatus.VerifySuccess(NativeMethods.OrtGetTensorShapeAndType(nativeOnnxValueMapKeys, out typeAndShape));
+
+                    elemType = NativeMethods.OrtGetTensorElementType(typeAndShape);
+                    if (typeAndShape != IntPtr.Zero)
+                    {
+                        NativeMethods.OrtReleaseTensorTypeAndShapeInfo(typeAndShape);
+                    }
+                    switch (elemType)
+                    {
+                        case TensorElementType.Int64:
+                            return DisposableNamedMapFromNativeMap<Int64, float>(string.Empty, nativeOnnxValueMapKeys, nativeOnnxValueMapValues);
+                        case TensorElementType.String:
+                            return DisposableNamedMapFromNativeMap<string, float>(string.Empty, nativeOnnxValueMapKeys, nativeOnnxValueMapValues);
+                        default:
+                            throw new NotSupportedException("Map of element type: " + elemType + " is not supported");
+                    }
+                default:
+                    throw new NotSupportedException("OnnxValueType : " + onnxValueType + " is not supported");
+            }
+        }
 
-        private static DisposableNamedOnnxValue NameOnnxValueFromNativeTensor<T>(string name, IntPtr nativeOnnxValue)
+        private static DisposableNamedOnnxValue DisposableNamedOnnxValueFromNativeTensor<T>(string name, IntPtr nativeOnnxValue)
         {
             NativeOnnxTensorMemory<T> nativeTensorWrapper = new NativeOnnxTensorMemory<T>(nativeOnnxValue);
             DenseTensor<T> dt = new DenseTensor<T>(nativeTensorWrapper.Memory, nativeTensorWrapper.Dimensions);
             return new DisposableNamedOnnxValue(name, dt, nativeTensorWrapper);
         }
 
+        private static DisposableNamedOnnxValue DisposableNamedMapFromNativeMap<K,V>(string name, IntPtr nativeOnnxValueKeys, IntPtr nativeOnnxValueValues)
```

Comment:
Nit: Consistency in naming functions -- may be better with DisposableNamedOnnxValueFromNativeMap

Reply:
Done.